sdks/java/io/mqtt/src/test/java/org/apache/beam/sdk/io/mqtt/MqttIOTest.java

```diff
@@ -57,70 +59,84 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MqttIOTest.class);
 
-  private static transient BrokerService broker;
+  private static transient BrokerService brokerService;
 
   private static int port;
 
-  @BeforeClass
-  public static void startBroker() throws Exception {
+  @Before
+  public void startBroker() throws Exception {
     LOGGER.info("Finding free network port");
     ServerSocket socket = new ServerSocket(0);
     port = socket.getLocalPort();
     socket.close();
 
-    LOGGER.info("Starting ActiveMQ broker on {}", port);
-    broker = new BrokerService();
-    broker.setUseJmx(false);
-    KahaDBPersistenceAdapter kahaDb = new KahaDBPersistenceAdapter();
-    kahaDb.setDirectory(new File("target/kahadb"));
-    broker.setPersistenceAdapter(kahaDb);
-    broker.addConnector(new URI("mqtt://localhost:" + port));
-    broker.start();
+    LOGGER.info("Starting ActiveMQ brokerService on {}", port);
+    brokerService = new BrokerService();
+    brokerService.setDeleteAllMessagesOnStartup(true);
+    brokerService.setPersistent(true);
+    brokerService.setDataDirectory("target/activemq-data/");
+    KahaDBStore kahaDBStore = new KahaDBStore();
+    kahaDBStore.setDirectory(new File("target/activemq-data/"));
+    brokerService.setPersistenceAdapter(kahaDBStore);
+    brokerService.setAdvisorySupport(false);
+    brokerService.setUseJmx(true);
+    brokerService.getManagementContext().setCreateConnector(false);
+    brokerService.setSchedulerSupport(true);
+    brokerService.setPopulateJMSXUserID(true);
+    TransportConnector mqttConnector = new TransportConnector();
+    mqttConnector.setName("mqtt");
+    mqttConnector.setUri(new URI("mqtt://localhost:" + port));
+    mqttConnector.setAllowLinkStealing(true);
```

Comment:
I recall that this in particular caused those hard-to-diagnose problems, why are we allowing link stealing, especially now that client ids are unique?